internals/secrethub/tpl/v2_test.go

```diff
@@ -1,23 +1,469 @@
+						},
+					},
+				},
+			},
+		},
+		"variable in secret path in middle": {
+			input: "{{path/to/${var}/secret}}",
+			expected: []node{
+				secret{
+					path: []node{
+						character('p'),
+						character('a'),
+						character('t'),
+						character('h'),
+						character('/'),
+						character('t'),
+						character('o'),
+						character('/'),
+						variable{
+							key: "var",
+						},
+						character('/'),
+						character('s'),
+						character('e'),
+						character('c'),
+						character('r'),
+						character('e'),
+						character('t'),
+					},
+				},
+			},
+		},
+		"variable with spaces": {
+			input: "${ var }",
+			expected: []node{
+				variable{
+					key: "var",
+				},
+			},
+		},
+		"secret with spaces": {
+			input: "{{ path/to/secret }}",
+			expected: []node{
+				secret{
+					path: []node{
+						character('p'),
+						character('a'),
+						character('t'),
+						character('h'),
+						character('/'),
+						character('t'),
+						character('o'),
+						character('/'),
+						character('s'),
+						character('e'),
+						character('c'),
+						character('r'),
+						character('e'),
+						character('t'),
+					},
+				},
+			},
+		},
+		"{ and } chars used": {
+			input: `{"key": "value"}`,
+			expected: []node{
+				character('{'),
+				character('"'),
+				character('k'),
+				character('e'),
+				character('y'),
+				character('"'),
+				character(':'),
+				character(' '),
+				character('"'),
+				character('v'),
+				character('a'),
+				character('l'),
+				character('u'),
+				character('e'),
+				character('"'),
+				character('}'),
+			},
+		},
+		"}} used outside secret tag": {
+			input: `{"a": {"b": "c"}}`,
+			expected: []node{
+				character('{'),
+				character('"'),
+				character('a'),
+				character('"'),
+				character(':'),
+				character(' '),
+				character('{'),
+				character('"'),
+				character('b'),
+				character('"'),
+				character(':'),
+				character(' '),
+				character('"'),
+				character('c'),
+				character('"'),
+				character('}'),
+				character('}'),
+			},
+		},
+		"$ used": {
+			input: `$12.50`,
+			expected: []node{
+				character('$'),
+				character('1'),
+				character('2'),
+				character('.'),
+				character('5'),
+				character('0'),
+			},
+		},
+		"escaped dollar": {
+			input: `\$`,
+			expected: []node{
+				character('$'),
+			},
+		},
+		"escaped dollar + bracket": {
+			input: `\${var}`,
+			expected: []node{
+				character('$'),
+				character('{'),
+				character('v'),
+				character('a'),
+				character('r'),
+				character('}'),
+			},
+		},
+		"escaped double bracket": {
+			input: `\{{ path }}`,
+			expected: []node{
+				character('{'),
+				character('{'),
+				character(' '),
+				character('p'),
+				character('a'),
+				character('t'),
+				character('h'),
+				character(' '),
+				character('}'),
+				character('}'),
+			},
+		},
+		"escaped backslash": {
+			input: `\\`,
+			expected: []node{
+				character('\\'),
```

Comment:
Haha this is confusing, because the `\` character needs to be escaped too in Golang. 😄

Reply:
Haha jup. With strings we can use backticks ``` ` ```, but as far as I know we don't have something similar for runes.